src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathSingleFilter.java

```diff
@@ -421,16 +423,19 @@
     /**
      * Define a Regular Expression matched against the file name associated with an audit event.
      */
+    @BeanPropertyType(PropertyType.REGEXP)
     private Pattern files;
```

Comment:
I'm not remembering why we have the difference between Pattern and RegExp.

Reply:
Me too. Let's discuss which of the two types to keep and make it consistent.

Reply:
https://checkstyle.org/property_types.html#Pattern
> This type represents a regular expression. The string representation is parsed using java.util.regex package. 

https://checkstyle.org/property_types.html#RegularExpression
> This type represents a regular expression. The string representation is parsed using java.util.regex package. 

I have seen `regexp` since the beginning in the history. Pattern came about in a064a46d47ac0f0c57163d455f3bc6d16f19f6c4 which was very recently. Commit message misleads on it being added on purpose but this probably all leads back to #8328 and https://github.com/checkstyle/checkstyle/issues/8328#issuecomment-651952351 .

Since there was some discussion on this, I am hesitant to say we can change it.

Reply:
@romani @strkkk Are you agree to change `regular expression` => `Pattern`?

Reply:
Ok, but let's do this in separate issue, there might be some tiny nuance.
But in referenced issues we did not add new types, we just declared them all. It might be a duplicate.

Reply:
done